backend/report/views.py

```diff
@@ -0,0 +1,289 @@
+            if len(finished_questionnaires_by_user) != 0:
+                finished_quetionnaires_df = pd.DataFrame(finished_questionnaires_by_user)
+                finished_quetionnaires_df['Date'] = finished_quetionnaires_df['finished_at'].dt.strftime('%Y-%m-%d')
+                finished_questionnaire_dates = list(finished_quetionnaires_df.Date.unique())
+                period_questionnaire_time = []
+                for date in finished_questionnaire_dates:
+                    list_finished_questionnaire_ids = finished_quetionnaires_df[finished_quetionnaires_df['Date'] == date]['questionnaire_id'].tolist()
+                    answered_questionnaires_by_user = Answer.objects.get_answers_time_by_user_in_given_period(list_finished_questionnaire_ids, user_id, date=date)
+                    if len(answered_questionnaires_by_user) != 0:
+                        answered_questionnaire_df = pd.DataFrame(answered_questionnaires_by_user).sort_values('created_at')
+                        answered_questionnaire_df = answered_questionnaire_df.drop_duplicates(['question_id__questionnaire_id'], keep='first')
+                        start_time_list = answered_questionnaire_df.created_at.to_list()
+                        finish_time_list = finished_quetionnaires_df[finished_quetionnaires_df['Date'] == date].finished_at.to_list()
+                        time_diff_list = [(fn - st).seconds for fn, st in zip(finish_time_list, start_time_list)]
+                        questionaire_time_per_day = {"date": date, "total_time (seconds)": sum(time_diff_list), "user_id": user_id}
+                        period_questionnaire_time.append(questionaire_time_per_day)
+                all_user_period_questionnaire_time.append(period_questionnaire_time)
+        if all_user_period_questionnaire_time:
+            final_user_period_questionnaire_time = []
+            for time in all_user_period_questionnaire_time:
+                for sample in time:
+                    final_user_period_questionnaire_time.append(sample)
+            df = pd.DataFrame(final_user_period_questionnaire_time)
+            list_user_questionnaire_avg_spending_time = df.groupby(["user_id"])['total_time (seconds)'].mean().to_list()
+            all_user_average_questionnaire_time = sum(list_user_questionnaire_avg_spending_time) / len(list_user_questionnaire_avg_spending_time)
+            avg_daily_questionnaire_time_df = df.groupby('date')['total_time (seconds)'].mean().reset_index(name='avg_questionnaire_time_daily (seconds)')
+            all_user_daily_avg_questionnaire_time = avg_daily_questionnaire_time_df.to_dict(orient='records')
+            data = {"average_daily_questionnaire_time (seconds)": all_user_average_questionnaire_time, "daily_questionnaire_time": all_user_daily_avg_questionnaire_time}
+        return Response(data=data, status=status.HTTP_200_OK)
```

Comment:
Hello, I am currently testing the apis from the FE side, I can confirm the other 4 apis for the all users statistics work, but the one with endpoint `/v1/report/all-users-avg-daily-questionnaire-time/2022-11-01/2023-01-31/` does not work. I got an error:
```
Internal Server Error: /v1/report/all-users-avg-daily-questionnaire-time/2022-11-01/2023-01-31/
Traceback (most recent call last):
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\django\views\decorators\csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\django\views\generic\base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\rest_framework\views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\rest_framework\views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\rest_framework\views.py", line 480, in raise_uncaught_exception
    raise exc
  File "C:\Users\Teddy Ferdinan\AppData\Local\pypoetry\Cache\virtualenvs\doccano-e82fD2CR-py3.8\lib\site-packages\rest_framework\views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "D:\Apps\doccano\backend\report\views.py", line 289, in get
    return Response(data=data, status=status.HTTP_200_OK)
UnboundLocalError: local variable 'data' referenced before assignment
[05/Jan/2023 12:05:59] "GET /v1/report/all-users-avg-daily-questionnaire-time/2022-11-01/2023-01-31/ HTTP/1.1" 500 101407
```

Would you please check it? 😅

Reply:
yes, thank you for pointing it out XD I forgot to add the covering for such case of no data in some APIs :sweat_smile:  I've updated XD